### PR TITLE
Add new appauth external browser support

### PR DESCRIPTION
**Pull request by wtrocki**
### Motivation

During the login SSO not always work immediately. 
Doing login in safari is usually recommended. 
I have added external browser support in order to perform more mobile friendly flow where login is happening in safari.

### CODE REVIEW

File: modules/auth/authenticator/OIDCAuthenticator.swift [hunk 1]
```diff
@@ -92,7 +93,18 @@ public class OIDCAuthenticator: Authenticator {
...
+        if(authConfig.useExternalUserAgent){
+            let externalUserAgent = OIDExternalUserAgentIOSCustomBrowser.customBrowserSafari();
+            return OIDAuthState.authState(byPresenting: byPresenting,
+                                          externalUserAgent: externalUserAgent,
+                                          callback:flowCallback)
+        }else {
+             return OIDAuthState.authState(byPresenting: byPresenting,
+                                           presenting: presenting,
+                                           callback: flowCallback)
+        }
```
Comment (wei-lee) [inline modules/auth/authenticator/OIDCAuthenticator.swift]: indentation seems to be off here?

File: modules/auth/config/AuthenticationConfig.swift [hunk 2]
```diff
@@ -18,8 +20,9 @@ public struct AuthenticationConfig {
         - redirectURL: the redirect URL for the developers app
         - minTimeBetweenJwksRequests: The minimum time, in minutes, between Json web key set requests. Default value is 1400 (1 day)
```
Comment (wei-lee) [inline modules/auth/config/AuthenticationConfig.swift]: should update the docs as well?

Comment (wtrocki) [reply modules/auth/config/AuthenticationConfig.swift]: Yes. Thanks for pointing that out.

**Comment by wtrocki** [comment]
Thanks for reviews. Rebased changes and formatted code using our swiftlint profile.

**Pull request merged**